### PR TITLE
RT-684-ExpressionChangedAfterItHasBeenCheckedError

### PR DESCRIPTION
**Pull request by yashpmohod**
## Overview
use `ChangeDetectorRef` to manipulate the tree.
image

## How it was tested
Tested using run showcase app locally using mock-fhir server and ngx-charts-on-fhir server.

## Checklist

- [x] The title contains a short meaningful summary
- [x] I have added a link to this PR in the Jira issue
- [x] I have described how this was tested
- [x] I have included screen shots for changes that affect the user interface
- [ ] I have updated unit tests
- [x] I have run unit tests locally
- [ ] I have updated documentation (including README)

### CODE REVIEW

File: projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ const max = 1650906227000; // Apr 25, 2022
 const min = 1578330227000; // Jan 6, 2020
 
 class MockConfigService {
-  timelineRange$ = of({ min, max });
+  timelineRange$ = of({ min, max }).pipe(delay(0));
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts:21]: why did you add a delay to the mock config service?

File: projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { MatInputModule } from '@angular/material/input';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { By } from '@angular/platform-browser';
 import { MatButtonToggleHarness } from '@angular/material/button-toggle/testing';
-import { of } from 'rxjs';
+import { delay, of } from 'rxjs';
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts:7]: unused import

**Review by semicolin — CHANGES_REQUESTED** [review]

File: projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts (hunk 2 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts:7]: Okay, @semicolin I will remove unused imports.

File: projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ const max = 1650906227000; // Apr 25, 2022
 const min = 1578330227000; // Jan 6, 2020
 
 class MockConfigService {
-  timelineRange$ = of({ min, max });
+  timelineRange$ = of({ min, max })
```
[the remark this answers is not in the document]
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/range-selector/range-selector.component.spec.ts:21]: Okay @semicolin

**Review by semicolin — APPROVED** [review]

**Pull request merged**